File: leetcode_python/14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: [str]) -> str:
        if not strs: return ""
        mi = len(strs[0])
        for s in strs[1:]: mi = min(mi, len(s))
        for i in range(mi):
            for j in range(len(strs) - 1):
                if strs[j][i] != strs[j + 1][i]:
                    return strs[j][:i]
        return strs[0][:i+1] if mi else ""


s = Solution()
print(s.longestCommonPrefix(["aaa", "aaa"]))

File: leetcode_java/14. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length == 0) return "";
        int min = strs[0].length(), i = 0;
        for(String s : strs) min = Math.min(min, s.length());
        for(; i < min; i++) {
            for(int j = 0; j < strs.length - 1; j++){
                if(strs[j].charAt(i) != strs[j+1].charAt(i)) return strs[j].substring(0,i);
            }
        }
        return min > 0 ? strs[0].substring(0, i) : "";
    }
}